File: Python/Priority queue/Maximal Score After Applying K Operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        # Use -i while pushing elements into the heap to simulate a max-heap using heapq (since heapq is a min-heap by default).
        score = 0
        pq = []
        for i in nums:
            heapq.heappush(pq, -i)
        while k > 0:
            k -= 1
            # Get the largest element by popping the smallest negative value
            maxele = -heapq.heappop(pq)
            score += maxele
            # Push the new value (ceiling of maxele/3) back into the heap as negative
            heapq.heappush(pq, -math.ceil(maxele / 3))
        return score

File: CPP/Priority queue/Maximal Score After Applying K Operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        long long score=0;
        priority_queue<int>pq;
        for(auto i:nums){
            pq.push(i);
        }
        for(int i=0;i<k;i++){
            int maxele=pq.top();
            score+=maxele;
            pq.pop();
            pq.push(ceil(maxele/3.0));
        }
        return score;
    }
};
